packages/grid/_modules_/grid/components/column-header-filter-icon.tsx

```diff
@@ -0,0 +1,232 @@
+//       margin: 'auto',
+//       display: 'block',
+//       width: 0,
+//       height: 0,
+//       borderStyle: 'solid',
+//     },
+//   },
+// }));
+
+export const ColumnFilterMenu: React.FC = () => {
+  // const classes = useStyles();
+  const apiRef = React.useContext(ApiContext);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [filterValue, setFilterValue] = React.useState('');
+  const [target, setTarget] = React.useState<HTMLElement | null>(null);
+  const [colDef, setColumn] = React.useState<ColDef | null>(null);
+  const icons = useIcons();
+  const filterIconElement = React.createElement(icons.columnFiltering!, {});
+
+  const hideTimeout = React.useRef<any>();
+  const hidePopper = React.useCallback(() => {
+    hideTimeout.current = setTimeout(() => setIsOpen(() => false), 50);
+  }, []);
+
+  const onColumnFilterClick = React.useCallback(
+    ({ element, column }) => {
+      setImmediate(() => clearTimeout(hideTimeout.current));
+      setIsOpen((p) => {
+        if (colDef == null || column.field !== colDef?.field) {
+          setFilterValue('');
+          return true;
+        }
+        return !p;
+      });
+      setTarget(element);
+      setColumn(column);
+    },
+    [colDef],
+  );
+
+  const onFilterChange = React.useCallback((event) => {
+    setFilterValue(event.target.value);
+  }, []);
+
+  React.useEffect(() => {
+    return apiRef?.current.subscribeEvent(COLUMN_FILTER_BUTTON_CLICK, onColumnFilterClick);
+  }, [apiRef, onColumnFilterClick]);
+
+  // const [arrowRef, setArrowRef] = React.useState<any>(null);
+  return (
+    <Popper
+      placement="bottom"
+      open={isOpen}
+      anchorEl={target}
+      disablePortal={false}
+      modifiers={{
+        flip: {
+          enabled: true,
+        },
+        preventOverflow: {
+          enabled: true,
+          boundariesElement: 'scrollParent',
+        },
+        // arrow: {
+        //   enabled: true,
+        //   element: arrowRef,
+        // },
+      }}
+    >
+      {/*<span className={classes.arrow} ref={setArrowRef} />*/}
+
+      <ClickAwayListener onClickAway={hidePopper}>
+        <Paper square>
+
+          <Tabs
+            value={0}
+            variant="fullWidth"
+            indicatorColor="primary"
+            textColor="primary"
+            scrollButtons="off"
+            aria-label="scrollable prevent tabs example"
+          >
+            <Tab icon={filterIconElement} />
+            <Tab icon={<ViewWeekIcon />} disabled />
+          </Tabs>
+          <TabPanel value={0} index={0}>
+            <div>
+              <TextField
+                id="standard-basic"
+                label={`${colDef?.headerName || colDef?.field}`}
+                placeholder={'Filter value'}
+                value={filterValue}
+                onChange={onFilterChange}
+              />
+              <IconButton color="primary" aria-label="Filter" component="span">
+                <Search />
+              </IconButton>
```

Comment:
Have you considered using a regular button instead? I was confused about the icon, I thought it was a presentational one.